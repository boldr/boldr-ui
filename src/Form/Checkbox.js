// @flow
import React from 'react';
import type { Node } from 'react';
import classNames from 'classnames';

import { createWrappedComponent } from '../util/boldrui';
import { StyleClasses } from '../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.CHECKBOX;

export type Props = {
  children: Array<Node>,
  disabled?: boolean,
  className?: string,
};

export function Checkbox(props: Props) {
  const wrapperClassName = classNames(BASE_ELEMENT, props.className);

  const { children, ...HTMLProps } = props;

  return (
    <label className={wrapperClassName} disabled={HTMLProps.disabled}>
      <input {...HTMLProps} type="checkbox" />
      {children}
    </label>
  );
}

export default createWrappedComponent(Checkbox);
