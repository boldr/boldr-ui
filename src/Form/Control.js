// @flow
import React from 'react';
import classNames from 'classnames';

import {
  getLoadingModifiers,
  removeLoadingProps,
  isLeft,
  isRight,
  createWrappedComponent,
} from '../util/boldrui';

import { isOption, getDomSafeProps } from '../util/helpers';
import { StyleClasses } from '../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.FORM_CONTROL;
export type Directions = 'left' | 'right';

export type Props = {
  hasIcons?: boolean,
  isExpanded?: boolean,
  tag?: string,
  className?: string,
};

const isDirection = isOption(isLeft, isRight);

const getModifier = modifier => {
  if (modifier === true) {
    return { 'has-icons-left has-icons-right': true };
  } else if (typeof modifier === 'string') {
    return isDirection(modifier) ? { [`has-icons-${modifier}`]: true } : {};
  } else if (Array.isArray(modifier)) {
    return modifier
      .map(str => str.toLowerCase().trim())
      .reduce(
        (init, option) => (isDirection(option) ? { ...init, [`has-icons-${option}`]: true } : init),
        {},
      );
  }

  return {};
};

export function Control({ tag = 'div', ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      ...getModifier(props.hasIcons),
      'is-expanded': props.isExpanded,
      ...getLoadingModifiers(props),
    },
    props.className,
  );
  const { hasIcons, isExpanded, ...rest } = props;

  const HTMLProps = getDomSafeProps(rest, removeLoadingProps);

  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(Control);
