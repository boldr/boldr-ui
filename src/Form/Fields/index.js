export { default as InputField } from './InputField';
export { default as SelectInput } from './SelectInput';
export { default as RadioField } from './RadioField';
export { default as RadioGroupField } from './RadioGroupField';
export { default as CheckboxField } from './CheckboxField';
export { default as CheckboxGroupField } from './CheckboxGroupField';
