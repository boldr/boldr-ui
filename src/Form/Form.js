/* @flow */
import * as React from 'react';
import styled from 'styled-components';

export type Props = {
  children: Array<React.Node>,
  isInline: boolean,
  handleSubmit: () => void,
  className?: string,
};

export type MetaProps = {
  error?: string,
  warning?: string,
  touched?: boolean,
};

const FormElement = styled.form`display: ${props => (props.isInline ? 'inline-block' : 'block')};`;

const Form = (props: Props) => {
  const { handleSubmit, className } = props;

  return (
    <FormElement className={className} onSubmit={handleSubmit} {...props}>
      {props.children}
    </FormElement>
  );
};

Form.defaultProps = {
  isInline: false,
};

export default Form;
