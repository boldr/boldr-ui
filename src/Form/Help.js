// @flow
import React from 'react';
import classNames from 'classnames';

import { getColorModifiers, removeColorProps, createWrappedComponent } from '../util/boldrui';
import { getDomSafeProps } from '../util/helpers';
import { StyleClasses } from '../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.FORM_HELP;
export type Props = {
  tag?: string,
  className?: string,
};

export function Help({ tag = 'p', ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      ...getColorModifiers(props),
    },
    props.className,
  );

  const HTMLProps = getDomSafeProps(props, removeColorProps);

  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(Help);
