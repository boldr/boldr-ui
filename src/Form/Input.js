// @flow
import React from 'react';
import classNames from 'classnames';

import {
  getColorModifiers,
  getSizeModifiers,
  getStateModifiers,
  removeColorProps,
  removeSizeProps,
  removeStateProps,
  createWrappedComponent,
} from '../util/boldrui';
import { combineModifiers, getDomSafeProps } from '../util/helpers';

import { StyleClasses } from '../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.INPUT;

export type Props = {
  className?: string,
  type?: string,
};

export function Input(props: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      ...combineModifiers(props, getColorModifiers, getSizeModifiers, getStateModifiers),
    },
    props.className,
  );

  const HTMLProps = getDomSafeProps(props, removeColorProps, removeSizeProps, removeStateProps);

  return <input {...HTMLProps} className={className} type={props.type || 'text'} />;
}

export default createWrappedComponent(Input);
