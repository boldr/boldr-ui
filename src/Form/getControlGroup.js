import React, { Component } from 'react';
import { string, bool } from 'prop-types';
import cx from 'classnames';

export default Control => {
  return class ControlGroup extends Component {
    static propTypes = {
      required: bool,
      helpDesc: string,
      label: string,
      className: string,
    };
    getControlInstance = () => {
      return this.control;
    };

    render() {
      const { required = false, helpDesc = '', label = '', className = '', ...props } = this.props;

      const showError = props.isTouched && props.error;
      const groupClassName = cx({
        'boldrui-form__control-group': true,
        'boldrui-form__control-group--active': props.isActive,
        'has-error': showError,
        [className]: true,
      });

      return (
        <div className={groupClassName}>
          <label className="boldrui-form__control-label">
            {required ? <em className="boldrui-form__required">*</em> : null}
            {label}
          </label>
          <div className="boldrui-form__controls">
            <Control {...props} ref={ref => (this.control = ref)} />
            {showError &&
              <p className="boldrui-form__error-desc">
                {props.error}
              </p>}
            {helpDesc &&
              <p className="boldrui-form__help-desc">
                {helpDesc}
              </p>}
          </div>
        </div>
      );
    }
  };
};
