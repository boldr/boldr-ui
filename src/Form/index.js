import FormCard from './FormCard';
import FormGroup from './FormGroup';
import Form from './Form';
// import Label from './Label';
export { Form, FormCard, FormGroup };

export { default as Checkbox } from './Checkbox';
export { default as Control } from './Control';
export { default as Help } from './Help';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Radio } from './Radio';
export { default as SelectInput } from './SelectInput';
export { default as TextArea } from './TextArea';
// Form - Field
export { default as FormField } from './Field/FormField';
export { default as FieldBody } from './Field/FieldBody';
export { default as FieldLabel } from './Field/FieldLabel';
export {
  TextFormField,
  TextAreaFormField,
  SelectFormField,
  CheckboxFormField,
  RadioFormField,
} from './ReduxFormFields';

export { default } from './Form';
