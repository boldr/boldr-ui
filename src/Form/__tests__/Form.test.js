import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Form from '../Form';

describe('<Form />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<Form />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
