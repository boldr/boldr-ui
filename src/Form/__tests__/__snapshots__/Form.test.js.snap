// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match the snapshot 1`] = `
<styled.form
  isInline={false}
/>
`;
